src/pages/instances/forms/DiskDeviceFormCustom.tsx

```diff
@@ -0,0 +1,260 @@
+    if (hasConflict) {
+      return deduplicateName(index + 1);
+    }
+    return candidate;
+  };
+
+  const rows: MainTableRow[] = [];
+
+  customVolumes.map((formVolume) => {
+    const index = formik.values.devices.indexOf(formVolume);
+
+    rows.push(
+      getConfigurationRowBase({
+        className: "no-border-top custom-disk-device-name",
+        configuration: (
+          <RenameDiskDeviceInput
+            name={formVolume.name}
+            index={index}
+            isReadOnly={isReadOnly}
+            setName={(name) =>
+              formik.setFieldValue(`devices.${index}.name`, name)
+            }
+          />
+        ),
+        inherited: "",
+        override: !isReadOnly && (
+          <DetachDiskDeviceBtn onDetach={() => removeDevice(index, formik)} />
+        ),
+      })
+    );
+
+    rows.push(
+      getConfigurationRowBase({
+        className: "no-border-top inherited-with-form",
+        configuration: (
+          <Label forId={`pool-${index}-pool`}>Pool / volume</Label>
+        ),
+        inherited: (
+          <div className="custom-disk-volume-source">
+            <div
+              className={classnames("mono-font", {
+                "u-truncate": !formik.values.readOnly,
+              })}
+              title={
+                formik.values.readOnly
+                  ? undefined
+                  : `${formVolume.pool} / ${formVolume.source ?? ""}`
+              }
+            >
+              <b>
+                {formVolume.pool} / {formVolume.source}
+              </b>
+            </div>
+            {!isReadOnly && (
+              <CustomVolumeSelectBtn
+                project={project}
+                setValue={(volume) => changeVolume(volume, formVolume, index)}
+                buttonProps={{
+                  id: `pool-${index}-pool`,
+                  appearance: "base",
+                  className: "u-no-margin--bottom",
+                }}
+              >
+                <Icon name="edit" />
+              </CustomVolumeSelectBtn>
+            )}
+          </div>
+        ),
+        override: "",
+      })
+    );
+
+    if (formVolume.path !== undefined) {
+      rows.push(
+        getConfigurationRowBase({
+          className: "no-border-top inherited-with-form",
+          configuration: (
+            <Label forId={`pool-${index}-path`}>Mount point</Label>
+          ),
+          inherited: isReadOnly ? (
+            <div className="mono-font">
+              <b>{formVolume.path}</b>
+            </div>
+          ) : (
+            <Input
+              id={`pool-${index}-path`}
+              onBlur={formik.handleBlur}
+              onChange={(e) => {
+                formik.setFieldValue(`devices.${index}.path`, e.target.value);
+              }}
+              value={formVolume.path}
+              type="text"
+              placeholder="Enter full path (e.g. /data)"
+              className="u-no-margin--bottom"
+            />
```

Comment:
I think this field should be `required`, to avoid the postponed feedback in case the user forgets to write the path and tries to save the changes:

![image](https://github.com/canonical/lxd-ui/assets/56583786/4226497f-5155-4d13-a261-7700d67c1233)

Reply:
Made the two labels required. We should add some logic to prevent submitting the forms with missing values in the future. But that is a bit more involved, so lets keep it as part of this issue https://github.com/canonical/lxd-ui/issues/485

Reply:
Yeah just marking the field as required doesn't do much indeed, we need to be able to properly validate these required fields so that they prevent the user from clicking the submit button if they contain invalid values. Let's remember this as a future improvement - out of scope for now.